pkg/policy/api/rule.go

```diff
@@ -143,6 +143,19 @@ type IngressRule struct {
 	// +optional
 	FromCIDR []CIDR `json:"fromCIDR,omitempty"`
 
+	// FromCIDRSet is a list of IP blocks which the endpoint subject to the
+	// rule is allowed to receive connections from in addition to FromEndpoints,
+	// along with a list of subnets contained within their corresponding IP block
+	// from which traffic should not be allowed.
```

Comment:
Needs a comment whether this conflicts with `FromCIDR` or whether both are allowed. If both are allowed, are they related with a logical AND or OR.

Reply:
Good point. The way I currently have the code structured,`FromCIDR` supersedes `FromCIDRSet`, e.g., if someone specifies `FromCIDR` to have `10.0.0.0/8`, and in the same rule specifies `FromCIDRSet` to have `10.0.0.0/8` except CIDR Prefix `10.96.0.0/12`, then the net result would be to allow all traffic from `10.0.0.0/8`, as the current implementation does not check whether there are overlaps across `FromCIDRSet` and `FromCIDR` in the rule. Is this acceptable?

Reply:
I think it should be an OR. Adding a prefix into `FromCIDR` or into `FromCIDRSet` with no exclude should be completely equivalent.
An exclude in a `FromCIDRSet` only reduces the allowed IP addresses from that particular `FromCIDRSet`'s prefix, and no other `FromCIDR` or `FromCIDRSet`. Overlaps among `FromCIDR`s and `FromCIDRSet`s is perfectly fine, since it should be an OR semantics.

Reply:
Agreed, OR or disallowed.

Reply:
Comments describing overlap and scope have been added to each CIDR-related API object.